### PR TITLE
[gha] work around very large logs sent to slack by prover daily tests.

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

If the file is too large slack rejects the message.   Truncate the message to 50 lines.   If you're reading more than that, just click the link.

### Have you read the [Contributing Guidelines on pull requests]

Yes

## Test Plan

CI - currently running: https://github.com/diem/diem/runs/3165951975?check_suite_focus=true

## Related PRs

None.

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

**Comment by rexhoffman** [comment]
Had to test here: https://github.com/diem/diem/runs/3166249117?check_suite_focus=true.  forgot to change the branch the first test was targeting.

File: .github/actions/slack-file/message_file_to_slack.sh [hunk 1]
```diff
@@ -58,7 +58,7 @@ fi
 
 if [ -e "${PAYLOAD_FILE}" ]; then
     jq -n \
-    --arg msg "$(cat "${PAYLOAD_FILE}")" \
+    --arg msg "$(head -n 50 < "${PAYLOAD_FILE}")" \
```
Comment (sausagee) [inline .github/actions/slack-file/message_file_to_slack.sh:61]: This looks fine by itself.  

Depending on how the caller job that generate this $PAYLOAD_FILE, we may run into a case where we cannot look at the full log.  For example, the caller job pipes the stdout and stderr to $PAYLOAD_FILE.  

I would suggest piping the stderr to stdout and teeing it to a file in the caller job. something like `2&>1 | tee ${PAYLOAD_FILE}`

**Review by sausagee — DISMISSED** [review]

File: .github/actions/slack-file/message_file_to_slack.sh (hunk 1 above)
Comment (rexhoffman) [reply .github/actions/slack-file/message_file_to_slack.sh:61]: Just saw @junkil-park  comment about this in slack on this as well.  I really dislike taking ownership of the job for them - no analysis of why the job misbehaved was attempted before asking me for help - I was going to push back before realizing I could fix their job in the action.   Anyway, I'll also open a pr to fix their build step for them.   I wont print large messages in the action as it'll just encourage future bad logging in jobs.

Comment (junkil-park) [reply .github/actions/slack-file/message_file_to_slack.sh:61]: @rexhoffman, I am sorry for the miscommunication. I never intended for you to take full ownership of our job. I didn't mean for you to fix our problem entirely on your own. I was simply asking for your insight on the issue or guidance on where I could start fixing the issue. I am more than willing to maintain our part. Thank you for your help. I appreciate it and again, I apologize for the miscommunication.

**Pull request merged**